### ISSUE TITLE
Suggestion: Better labels for git issue priority

### ISSUE THREAD

**Issue by aubergine10** [DoESLiverpool/somebody-should#477]
A somewhat unusual use for milestones, but if we create 'priority milestones' it would help separate out the issue priority from other categorisations.

Milestones could be:

* Urgent
* Soon
* Whenever
* Ponder

Anyone got any objections to me trialling this next week?

We could try both approaches in parallel for a while to see how it works out in practice before making final decision.

**Comment by amcewen** [comment]
Go for it.  

The current system with the labels starting with numbers was to let us (ab)use HuBoard to make it easier to move things around and reprioritise them when we did a triage session.

However, I don't think we've ever done a lot of reprioritising things, so losing that isn't likely to make much difference.

We also haven't done a lot of triage sessions (two?  three maybe?  In how many years?).   I think that's the bigger problem, and lets all sorts of issues languish :-/

The "Ponder" category is a good one.  I'd intended adding a "Waiting for a champion" label so we could close some of the older issues that seem good ideas but no-one wants to run with, but add that label so they could easily be found as a group to look through in the future.

I don't know if that would work to clear out some of the clutter that would get shifted into the Ponder milestone?  They're not quite the same thing (I think) but thought I'd throw it in the mix.  Feel free to implement whatever you think most sensible though :-)

(And, btw, am appreciating your closing old issues and general poking of things, @aubergine10 :100:)

**Comment by aubergine10** [comment]
I'm not sure that it's working well to have both "stuff needs doing" being mixed with "stuff that's fun to do". It makes it difficult to get a handle on mission critical stuff. I can't see the wood for the trees.

I'll take a look at HuBoard first before doing anything - can I login to the DoES instance of HuBoard? (my email is guy.fraser1@gmail.com if required).

As for labels, I think we're already drowned in them to the point where it's actually getting difficult to work with labels. That was one of my motives for using milestones to denote priority - label culling.

I like the idea of having some way to denote "needs a champion" - in particular it would be good way to highlight things on the website as well (as part of #weeknotes).

I'll be doing another wade through existing tickets either today or tomorrow to prod those that look stale and hope to have a load more closed by next #makernight.

**Comment by DoESsean** [comment]
Given that I'm basically going through all of my alerts and saying 'Don't delete this!', I just wanted to take this opportunity to echo what @amcewen said - that you're taking the time to go through all these and try to sort the wheat from the chaff is very much appreciated, and something that we should be better at doing (maybe I should open an issue...)

A new tag for things that are long term rather than mission critical could be useful (although I think, given our current naming scheme, that 'Holding Out For A Hero' is better than 'Waiting For A Champion,' Adrian), especially if that means that they can be closed and recalled easily later on.

**Comment by SayBeano** [comment]
I'm a massive fan of MoSCoW beause it has a clear term to apply for not doing something (i.e. closing the thing) and fewer terms for backlog items / wishful thinking.

- Must
- Should
- Could
- Won't

https://en.wikipedia.org/wiki/MoSCoW_method

**Comment by ajlennon** [comment]
I like this a lot. Thanks @SayBeano

**Comment by DoESsean** [comment]
I'm now incredibly annoyed by the fact that it should be MuSCoW, but that
doesn't mean anything so they've changed it to MoSCoW hoping no-one will
notice.

**Comment by huffeec** [comment]
Technically shouldn't it be Mushcowo, which is also hipster for beefburger..

**Comment by amcewen** [comment]
"And that was how, back in 2017, DoES Liverpool switched from being a co-working and makerspace and founded the now-legendary chain of 3d-printed pizza and laser-grilled burger restaurants..."

**Comment by amcewen** [comment]
I know @SayBeano's suggestion makes lots of sense, but I'm also digging @DoESsean's far better punning :-D

@aubergine10 Point taken on the overwhelming-ness of lots of labels, but I think we should use them liberally.  They're especially useful as a lightweight categorisation system for issues that have been closed, which is useful because the closed issues are effectively the DoES Liverpool memory - slightly disorganised, but that's because we're optimising for the common case of them not needing to be looked at again, under the assumption that someone looking through them is doing so for a reason (they're trying to find the details of how we last fixed the 3d printer, for example) and so have an incentive to overcome that barrier.  (The alternative being that there's a much higher workload done when the issue is initially raised and closed, and IME there's no incentive for the work to be done then - plus a load of it is wasted, because lots of the issues won't be looked at again)

I think we can do a much better job of trying to explain how things work to newcomers.  Hopefully that'd make things a bit more understandable?  The intro is pretty basic at the minute, but that was mostly because I wanted to see if this system took off before spending lots of time on it.  I think we can probably say that it has taken off now...

**Comment by stevesparrow** [comment]
So, what should we do.

I love the idea of simple prioritising and sorting.  Do we have a consensus?

**Comment by aubergine10** [comment]
Well, we can parallel run (let's try the MSCW tagging first) - I'll do the updates on Friday both in github and HuBoard and then we have next week to stroke our beards and ponder whether it's a step in the right direction.

**Comment by ajlennon** [comment]
If we're looking at MoSCoW what about Organisers' actions being S.M.A.R.T. too? (or perhaps they already are?)

- Specific – target a specific area for improvement.
- Measurable – quantify or at least suggest an indicator of progress.
- Achievable – state what results can realistically be achieved, given available resources.
- Responsible – specify who will do it.
- Time-related – specify when the result(s) can be achieved.

https://en.wikipedia.org/wiki/SMART_criteria

**Comment by aubergine10** [comment]
I'll be starting on this task this week, but first wanted to ask if anyone is already familiar with any of the following - if so, are changes here on github likely to bugger up these things?

* #Weeknotes on the website, google group email, etc -- public facing, widely used
* [DoES Equipment Status](http://status.doesliverpool.com/) webapp -- public facing, but probably rarely used?
* HuBoard -- mostly internal use, and not used much from what I hear
* Any other integrations that might be affected? -- let me know!

**Comment by stevesparrow** [comment]
While this is going on, I wanted to draw attention again to the current priority labels.

1 - You're all I need to get by
2 - The time is now.
3 - Wouldn't it be nice.

I'm mentioning this as there are a lot of long term 'wouldn't it be nice' actions that do still need addressing, now marked as stale today and due to be deleted, that still are long terms aims or needs for us.  I've done a few.  If anyone has time to go through and save the ongoing issues I'd be grateful please.

Thanks
Steve





Thanks
Steve

**Comment by aubergine10** [comment]
Those issues were so far beyond "stale" they're a biohazard - and some have probably started to evolve in to new forms of intelligent life.

Seriously though, if an issue hasn't been updated for over 2 years then the chances are that it either a) isn't necessary or b) won't ever get done. If there are plans to do something with those issues, they should be given the S.M.A.R.T. treatment so they don't linger on for another 2 years.

**Comment by amcewen** [comment]
I like the addition of the "Stale?" label to help organise/triage some of the older issues.

When closing things I think it's also useful to note why they were closed, again with labels where appropriate.  There's already a "wontfix" label for things we decide are problems but ones we don't (currently) believe are worth bothering with.  I've created (from @DoESsean's suggestion) "Holding out for a hero" for issues that (a) are problems, (b) would be nice to fix, but (c) no-one has run at for a while.

My thinking behind that is stop us losing information about bits of kit.  For example, if/when someone wants to get the Proxxon CNC mill running well, it'll help them to find what we know is wrong with it (rather than having to look through _all_ the things that _have been_ wrong with it.

We do also need to better document what the labels are (especially when they've got slightly cryptic song-pun names :-D), so I've spun that out as its own issue - #487.

**Comment by stevesparrow** [comment]
Thanks Guy.  I'm all for removing older issues once it's established that they aren't relevant any longer. It's a great thing to look at the issues, establish what's needed and look at a different treatment for them as you suggest, deleting those which aren't.  It's a good suggestion.  Why don't we find a way to support you to do that?

Mostly, I'd like to do that before the 90 odd emails you've generated becomes 180 and then doubles again :) as at the moment each one, though very useful requires someone else to follow-on, check and do that themselves, which can easily be more time than is available - why I was urging a little more caution.

I had a chat to Adrian, who's at Maker night this evening if you are - and has some ideas. :)

Cheers
Steve

**Comment by aubergine10** [comment]
Yup, I'll be at maker night :)

I've no idea how to reduce the amount of emails when doing bulk updates; so far my approach has been to limit to roughly 25 stale tickets per batch (the number will dwindle as time goes on).

**Comment by stevesparrow** [comment]
Thanks.  It's not the emails so much, but what has to be done with them.  Adrian has some ideas/plans around a triage though.

**Comment by aubergine10** [comment]
One thing that would definitely help is if a label could be automatically added to all new issues, specifically the "Priority 0: Triage" label.

**Comment by aubergine10** [comment]
Current state of prioritisation labels:

![region capture 1](https://cloud.githubusercontent.com/assets/1386719/25666574/d0f7d8d2-3018-11e7-960d-180228d6f12c.png)

As we've got so many labels, I've grouped each category - for example, all rooms are prefixed "Room:", all tools "Tool:" and so on. With the exception of priority labels which have unique colour per label, all the other labels have been colour coded based on prefix (all tools green, all rooms blue, etc).

There are still some stragglers that I'm pondering over, but things are starting to look a bit more organised.
